gateway/oauth_manager_test.go

```diff
@@ -374,6 +376,87 @@ func TestAPIClientAuthorizeAuthCode(t *testing.T) {
 	})
 }
 
+func TestAPIClientAuthorizeRequestWithPKCE(t *testing.T) {
+	ts := StartTest()
+	defer ts.Close()
+
+	spec := LoadAPI(buildTestOAuthSpec(true))[0]
+
+	pID := CreatePolicy(func(p *user.Policy) {
+		p.ID = "TEST-4321"
+		p.AccessRights = map[string]user.AccessDefinition{
+			"test": {
+				APIID: "test",
+			},
+			"abc": {
+				APIID: "abc",
+			},
+		}
+	})
+
+	var redirectURI string
+	// If separator is not set that means multiple redirect uris not supported
+	if config.Global().OauthRedirectUriSeparator == "" {
+		redirectURI = "http://client.oauth.com"
+
+		// If separator config is set that means multiple redirect uris are supported
```

Comment:
Does this test's output depend on who runs it?

Reply:
No it wont make any difference - this is basically the boilerplate code for intitialising an Oauth client for tests but with the client ID set to an empty string like it would be when doing this via the Gateway API.